theano/scalar/basic.py

```diff
@@ -199,7 +199,11 @@ def filter(self, data, strict=False, allow_downcast=None):
                 type(data), data, self.dtype), e)
 
     def values_eq_approx(self, a, b, tolerance=1e-4):
-        return abs(a - b) <= ((abs(a) + abs(b)) * tolerance)
+        # The addition have risk of overflow especially with [u]int8
+        diff = a - b
+        if diff == 0:
+            return True
+        return abs(diff) <= ((abs(a) + abs(b)) * tolerance)
```

Comment:
There might still be an overflow if the difference is non-zero. What about : 
return abs(diff) <= (abs(a) * tolerence) + (abs(b) * tolerance)
Since tolerance is likely to be floating point, I think this should be fine.